aiohue/v2/__init__.py

```diff
@@ -0,0 +1,200 @@
+"""Control a Philips Hue bridge with V2 API."""
+from __future__ import annotations
+
+import asyncio
+from contextlib import asynccontextmanager
+import logging
+from types import TracebackType
+from typing import Callable, Generator, List, Optional, Type
+
+import aiohttp
+from aiohttp import ClientResponse
+
+from ..errors import Unauthorized, raise_from_error
+from .controllers.events import EventCallBackType, EventStream
+
+from .controllers.config import ConfigController
+from .controllers.sensors import SensorsController
+from .controllers.lights import LightsController
+from .controllers.groups import GroupsController
+from .controllers.scenes import ScenesController
+from .controllers.devices import DevicesController
+
+
+class HueBridgeV2:
+    """Control a Philips Hue bridge with V2 API."""
+
+    def __init__(
+        self,
+        host: str,
+        app_key: str,
+        websession: aiohttp.ClientSession | None = None,
+    ) -> None:
+        """
+        Initialize the Bridge instance.
+
+        Parameters:
+            `host`: the hostname or IP-address of the bridge as string.
+            `app_key`: provide the hue appkey/username for authentication.
+            `websession`: optionally provide a aiohttp ClientSession.
+        """
+        self._host = host
+        self._app_key = app_key
+        self._websession = websession
+        self._websession_provided = websession is not None
+
+        self.logger = logging.getLogger(f"{__package__}[{host}]")
+        self._events = EventStream(self)
+        # all resource controllers
+        self._config = ConfigController(self)
+        self._devices = DevicesController(self)
+        self._lights = LightsController(self)
+        self._scenes = ScenesController(self)
+        self._groups = GroupsController(self)
+        self._sensors = SensorsController(self)
+
+    @property
+    def bridge_id(self) -> str | None:
+        """Return the ID of the bridge we're currently connected to."""
+        return self._config.bridge_id
+
+    @property
+    def host(self) -> str:
+        """Return the hostname of the bridge."""
+        return self._host
+
+    @property
+    def events(self) -> EventStream:
+        """Return the EventStream controller."""
+        return self._events
```

Comment:
I guess we could just store this as `self.events` in constructor and we can skip these properties.

Reply:
I'm a bit autistic maybe but should a consumer of the lib be able to (accidentally) change one of those core attributes ? Maybe just a pattern thing but I've made all controllers as private vars and consumers are only able to get it, not change.

If you think it's not that kind of a problem I'll adjust and just make them all public vars